sql/2021/capabilities/desktop/absolute_orientation_sensor.sql

```diff
@@ -0,0 +1,21 @@
+CREATE TEMP FUNCTION
+getFuguAPIs(DATA STRING)
+RETURNS ARRAY<STRING>
+LANGUAGE js AS '''
+const $ = JSON.parse(data);
+return Object.keys($);
+''';
+SELECT
+  fuguAPI,
+  url,
+  payload
+FROM
+  `httparchive.pages.2021_07_01_desktop`,
+  UNNEST(getFuguAPIs(JSON_QUERY(payload, '$."_fugu-apis"'))) AS fuguAPI
+WHERE
+  JSON_QUERY(payload, '$."_fugu-apis"') != "[]" AND
+  fuguAPI = "Absolute Orientation Sensor"
+GROUP BY
+  fuguAPI,
+  url,
+  payload;
```

Comment:
Do we need to separate out desktop and mobile? Could we not just do this like we usually do?:

```suggestion
SELECT
  _TABLE_SUFFIX as client
  fuguAPI,
  url,
  payload
FROM
  `httparchive.pages.2021_07_01_*`,
  UNNEST(getFuguAPIs(JSON_QUERY(payload, '$."_fugu-apis"'))) AS fuguAPI
WHERE
  JSON_QUERY(payload, '$."_fugu-apis"') != "[]" AND
  fuguAPI = "Absolute Orientation Sensor"
GROUP BY
  client,
  fuguAPI,
  url,
  payload;
```

Also do we really want to return each url AND the full payload at that - not even restricting it to the '$."_fugu-apis"' payload? Aren't counts better? Could always give a sample of URLS if you want (have a look at [this example of how to do that](https://github.com/HTTPArchive/bigquery/blob/master/sql/histograms/htmlElementPopularity.sql).

Also it wouldn't run cause you've capitalised `data` on line 2 so need to revert that.

Finally I do wonder if it would be easier to run this as one big query and pivot on the fuguAPI in sheets? Might be easier...

Reply:
Muuuuuch nicer in latest version!

Reply:
Thanks for the pointers :-)